cvat-ui/src/components/task-page/details.tsx

```diff
@@ -130,6 +141,26 @@ export default class DetailsComponent extends React.PureComponent<Props, State>
         this.mounted = false;
     }
 
+    private onChangeRepoValue = (value: string): void => {
+        const { taskInstance } = this.props;
+        this.setState({ changedRepo: value });
```

Comment:
First we can get the current state value of ``repository`` state and keep it in memory.
Then do setState with the new ``repository`` value. At the same time I would add ``repositoryUpdating`` flag (to the state). When the flag is true, I would show a loading icon near the field, and disable all the repository fields (Because updating can take long time)
Finally we need to start server request.
In ``then/catch`` set appropriate state for ``repository`` and ``updatingRepository``. For example in ``catch`` you can revert the previous ``repository`` value (request failed)

What do you think?

Reply:
@bsekachev , the thing is, that according to current logic, there is no immediate initialization. 
Initialization happens only after other events like get/push etc. changeRepo request triggers only changes of appropriate rows in DB. That is why there is no long-run jobs.

Reply:
Sorry, I didn't catch your idea. There is no long-run server jobs, but there is a server request and its routing/handling can take time (quite long btw for highly loaded servers like cvat.org, for example), so while this processing a user should see an updated field I believe (just strange to see unedited value after a user finishes editing, do not you think?). And better if a user saw some ``loading`` status to understand that there changes weren't actually applied yet.

What is more, the request may be failed, in this case we should revert fiels (otherwise the user sees data which are not real, I mean they are not saved on the sever), it is why I suggested to handle with ``catch`` additionally.